core/src/gl/renderState.cpp

```diff
@@ -21,4 +21,12 @@ namespace RenderState {
         glClearColor(0.3, 0.3, 0.3, 1.0);
     }
     
+    void invalidateAllStates() {
+        blending.invalidate();
+        culling.invalidate();
+        depthTest.invalidate();
+        depthWrite.invalidate();
+        blendingFunc.invalidate();
+    }
+    
```

Comment:
I mean one could set the flag 'invalidate' in onContextDestroyed on then check in render whether to re-init the state.

... below code misses one m_valid = true, btw